awx/ui/src/screens/TopologyView/MeshGraph.js

```diff
@@ -0,0 +1,434 @@
+      { source: "aapc1.local", target: "aape1.local" },
+      { source: "aapc1.local", target: "aape2.local" },
+
+      { source: "aapc2.local", target: "aapc3.local" },
+      { source: "aapc2.local", target: "aape1.local" },
+      { source: "aapc2.local", target: "aape2.local" },
+
+      { source: "aapc3.local", target: "aape1.local" },
+      { source: "aapc3.local", target: "aape2.local" },
+
+      { source: "aape3.local", target: "aaph1.local" },
+      { source: "aape3.local", target: "aaph2.local" },
+
+      { source: "aape4.local", target: "aaph3.local" },
+
+      { source: "aaph1.local", target: "aapc1.local" },
+      { source: "aaph1.local", target: "aapc2.local" },
+      { source: "aaph1.local", target: "aapc3.local" },
+
+      { source: "aaph2.local", target: "aapc1.local" },
+      { source: "aaph2.local", target: "aapc2.local" },
+      { source: "aaph2.local", target: "aapc3.local" },
+
+      { source: "aaph3.local", target: "aaph1.local" },
+      { source: "aaph3.local", target: "aaph2.local" }
+    ]
+  };
+
+  const draw = useCallback(() => {
+    const margin = 80;
+    const getWidth = () => {
+      let width;
+      // This is in an a try/catch due to an error from jest.
+      // Even though the d3.select returns a valid selector with
+      // style function, it says it is null in the test
+      try {
+        width =
+          parseInt(d3.select(`#chart`).style('width'), 10) - margin || 700;
+      } catch (error) {
+        width = 700;
+      }
+
+      return width;
+    };
+    const width = getWidth();
+    const height = 600;
+    const defaultRadius = 6;
+    const highlightRadius = 9;
+
+    const zoom = d3
+      .zoom()
+      .scaleExtent([1, 8])
+      .on('zoom', function (event) {
+        svg.selectAll('.links, .nodes').attr('transform', event.transform);
+      });
+
+    /* Add SVG */
+    d3.selectAll(`#chart > *`).remove();
+
+    const svg = d3
+      .select('#chart')
+      .append('svg')
+      .attr('width', `${width + margin}px`)
+      .attr('height', `${height + margin}px`)
+      .append('g')
+      .attr('transform', `translate(${margin}, ${margin})`)
+      .call(zoom);
+
+    const color = d3.scaleOrdinal(d3.schemeCategory10);
+    const graph = data;
+
+    const simulation = d3
+      .forceSimulation()
+      .force(
+        'link',
+        d3.forceLink().id(function (d) {
+          return d.hostname;
+        })
+      )
+      .force('charge', d3.forceManyBody().strength(-350))
+      .force(
+        'collide',
+        d3.forceCollide(function (d) {
+          return d.node_type === 'execution' || d.node_type === 'hop'
+            ? 75
+            : 100;
+        })
+      )
+      .force('center', d3.forceCenter(width / 2, height / 2));
+
+    const link = svg
+      .append('g')
+      .attr('class', `links`)
+      .selectAll('path')
+      .data(graph.links)
+      .enter()
+      .append('path')
+      .attr('class', (d, i) => `link-${i}`)
```

Comment:
We'll need something more explicit here in order to be able to verify the link between nodes in UI tests, something like `${hostSource}-${hostTarget}` or `${hostSourceId}-${hostTargetId}`